contracts/src/bridge/Inbox.sol

```diff
@@ -298,8 +298,8 @@ contract Inbox is DelegateCallAware, PausableUpgradeable, IInbox {
         bytes calldata data
     ) external payable virtual override whenNotPaused returns (uint256) {
         // ensure the user's deposit alone will make submission succeed
-        if (msg.value < maxSubmissionCost + l2CallValue)
-            revert InsufficientValue(maxSubmissionCost + l2CallValue, msg.value);
+        uint256 ethNeeded = maxSubmissionCost + l2CallValue + gasLimit * maxFeePerGas;
+        if (msg.value < ethNeeded) revert InsufficientValue(ethNeeded, msg.value);
```

Comment:
would cost some gas to have the extra variable here
```suggestion
        if (msg.value < (maxSubmissionCost + l2CallValue + gasLimit * maxFeePerGas))
            revert InsufficientValue(ethNeeded, msg.value);
```

Reply:
I assumed it'd just be at the bottom of the stack and not cause any issues. I'm happy to change it but I'm curious why it costs extra gas.

Reply:
I assumed it'd just be at the bottom of the stack and not cause any issues. I'm happy to change it but I'm curious why it costs extra gas.

Reply:
You would need to PUSH the intermediate value and POP later, along with some DUP and SWAP. My experience is that usually cost ~10 gas, definitely insignificant but also easy for us to optimize here.

Reply:
Fixed